sdk/metric/metricdata/data.go

```diff
@@ -137,6 +137,70 @@ type HistogramDataPoint[N int64 | float64] struct {
 	Exemplars []Exemplar[N] `json:",omitempty"`
 }
 
+// ExponentialHistogram represents the histogram of all measurements of values from an instrument.
+type ExponentialHistogram[N int64 | float64] struct {
+	// DataPoints are the individual aggregated measurements with unique
+	// Attributes.
+	DataPoints []ExponentialHistogramDataPoint[N]
+	// Temporality describes if the aggregation is reported as the change from the
+	// last report time, or the cumulative changes since a fixed start time.
+	Temporality Temporality
+}
+
+func (ExponentialHistogram[N]) privateAggregation() {}
+
+// ExponentialHistogramDataPoint is a single histogram data point in a timeseries.
+type ExponentialHistogramDataPoint[N int64 | float64] struct {
+	// Attributes is the set of key value pairs that uniquely identify the
+	// timeseries.
+	Attributes attribute.Set
+	// StartTime is when the timeseries was started.
+	StartTime time.Time
+	// Time is the time when the timeseries was recorded.
+	Time time.Time
+
+	// Count is the number of updates this histogram has been calculated with.
+	Count uint64
+	// Min is the minimum value recorded. (optional)
+	Min Extrema[N]
+	// Max is the maximum value recorded. (optional)
+	Max Extrema[N]
+	// Sum is the sum of the values recorded.
+	Sum N
+
+	// Scale describes the resolution of the histogram. Boundaries are
+	// located at powers of the base, where:
+	//
+	//   base = (2^(2^-scale))
+	Scale int32
+	// ZeroCount is the count of values that are rounded to zero. This bucket
+	// stores values that cannot be expressed using the standard exponential formula.
+	ZeroCount uint64
+
+	// PositiveOffset is the positive bucket index of the first entry in the
+	// PositiveCounts slice.
+	PositiveOffset int32
+	// PositiveCounts is an slice where PositiveCounts[i] carries the count of
+	// the bucket at index (offset+i). PositiveCounts[i] is the count of values
+	// greater than base^(offset+i) and less than or equal to base^(offset+i+1).
+	PositiveCounts []uint64
+
+	// NegativeOffset is the positive bucket index of the first entry in the
+	// NegativeCounts slice.
+	NegativeOffset int32
+	// NegativeCounts is an slice where NegativeCounts[i] carries the count of
+	// the bucket at index (offset+i). NegativeCounts[i] is the count of values
+	// greater than base^(offset+i) and less than or equal to base^(offset+i+1).
+	NegativeCounts []uint64
+
+	// ZeroThreshold is the width of the zero region. Where the zero region is
+	// defined as the closed interval [-ZeroThreshold, ZeroThreshold].
+	ZeroThreshold float64
```

Comment:
The OTLP defines this dynamically as `zero_count/ count` right?

https://github.com/open-telemetry/opentelemetry-proto-go/blob/a8511867c40a8fedbc31ceeede8234d03d645b7a/otlp/metrics/v1/metrics.pb.go#L1325-L1333

Should we do the same?

Reply:
Nothing like this is defined in the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/metrics/data-model.md#zero-count-and-zero-threshold

Reply:
The zero count is because the buckets, both positive and negative, represent widths that are all some power of 2 e.g. `scale=3`  a bucket at index 5 is values `[ 2^(5/8), 2^(6/8) )`.  So there is no way to represent a bucket that starts at 0.  
The zero-width defines the upper bound of the zero bucket, as an optimization.  Because we can actually represent numbers smaller than the smallest non-zero floating point.